pkg/scd/store/cockroach/store.go

```diff
@@ -13,33 +15,85 @@ var (
 	DefaultClock = clockwork.NewRealClock()
 )
 
-// Store is an implementation of dss.Store using
-// Cockroach DB as its backend store.
+// Store is an implementation of scd.Store using
+// a CockroachDB transaction.
 type Store struct {
-	*cockroach.DB
+	tx     *sql.Tx
+	ctx    context.Context
 	logger *zap.Logger
 	clock  clockwork.Clock
 }
 
-// NewStore returns a Store instance connected to a cockroach instance via db.
-func NewStore(db *cockroach.DB, logger *zap.Logger) *Store {
+// Transaction is an implementation of scd.Transaction using
+// a CockroachDB transaction.
+type Transaction struct {
+	tx     *sql.Tx
+	logger *zap.Logger
+	ctx    context.Context
+	clock  clockwork.Clock
+}
+
+// Implement store.Transaction interface
+func (t *Transaction) Store() (scdstore.Store, error) {
```

Comment:
can we have this return an interface, which will help for testing, and allow us to keep this modular

Reply:
Could you elaborate?  `scdstore.Store` is an [interface](https://github.com/interuss/dss/pull/307/files#diff-2e50608ee73cee7175e96d52bd2bd2eeL51)

Reply:
ahh we're overloading store/cockroach/repo a lot.. there's a cockroach store which is a struct which I thought we were returning here